API/CopilotService/__init__.py

```diff
@@ -25,8 +25,14 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
 
     try:
         result = []
+        error = None
         if service_type == ServiceType.KNOWLEDGE_SEARCH:
             result = knowledge_search(question, top_num)
+            if len(result) > 0 and result[0]['score'] < float(os.environ.get('KNOWLEDGE_QUALITY_THRESHOLD', "1.0")):
+                error = f"Knowledge quality is too low, score: {result[0]['score']}"
+                result = []
+            elif len(result) == 0:
+                error = "No knowledge found"
```

Comment:
In fact, I am not sure if the reason of none high-quality data should be defined as error message.
If the caller doesn't care about the specific details of the reason for none data, it's just that we need to know this information when troubleshooting, could we replace it with printing log?